phusion-openldap-builder/Dockerfile

```diff
@@ -0,0 +1,45 @@
+ARG PHUSION_VERSION
+ARG OPENSSL_BUILDER_TAG
+ARG OPENLDAP_VERSION
+
+
+FROM openssl-builder:$OPENSSL_BUILDER_TAG as OpenSSL-builder
+
+FROM phusion/baseimage:$PHUSION_VERSION
+ENV PATH="/usr/local/ssl/bin:${PATH}" \
+    LD_LIBRARY_PATH="/usr/local/ssl/lib" \
+    OPENSSL_FIPS=1
+
+COPY --from=OpenSSL-builder /usr/local/ssl/ /usr/local/ssl/
+
+# Install the dependencies for OpenLDAP with OpenSSL
+RUN DEBIAN_FRONTEND=noninteractive  apt-get update -y \
+    && DEBIAN_FRONTEND=noninteractive apt-get install -y devscripts wget \
+    && DEBIAN_FRONTEND=noninteractive apt-get build-dep -y libldap2-dev
+
+# OpenLDAP 2.4.46 is the earliest version to support OpenSSL 1.1.1,
+# which is included with phusion/baseimage:0.11
+ARG OPENLDAP_VERSION
+ARG openldap_base_url=https://launchpad.net/ubuntu/+archive/primary/+sourcefiles/openldap/${OPENLDAP_VERSION}+dfsg-5ubuntu1.2
+
+RUN cd /root \
+    #
+    # The openldap source in the debian packages for 0.11 are for 2.4.45, so we pull
+    # the source for 2.4.46 using direct links:
+    && wget -q ${openldap_base_url}/openldap_${OPENLDAP_VERSION}+dfsg.orig.tar.gz \
```

Comment:
Is some checksum available for this file?
If so I think that we should validate this file using the checksum, see the [ruby builder](https://github.com/cyberark/conjur-base-image/blob/75f36dfe466466b4d7fcf07f029258a5667d8058/phusion-ruby-builder/Dockerfile#L14) example.

Reply:
The also for the file downloaded using `curl`.
By the way what is purpose of each file and why each one is downloaded using different tool?

Reply:
I've added the checksums and switched them both to use `wget`.